### PR TITLE
Add devenv

### PR DESCRIPTION
**Pull request by shyim**
Adds [devenv](https://devenv.sh/) to the docs as installation method

### CODE REVIEW

**Comment by fschmtt** [comment]
Open tasks:
- [ ] PhpStorm setup: path to PHP and other binaries
- [ ] PhpStorm setup: connect to DB using DB Tool Window (alternative to using adminer)
- [ ] `env.` file: use `127.0.0.1` instead of `localhost` in `DATABASE_URL`
- [ ] Add section "Known issues": Large repo causes slow shell (https://github.com/cachix/devenv/issues/257)
- [ ] Guide on `devenv gc`

File: SUMMARY.md [hunk 1]
```diff
@@ -115,6 +115,7 @@
   * [Installation from scratch](guides/installation/from-scratch.md)
   * [Valet+](guides/installation/valet.md)
   * [MAMP](guides/installation/mamp.md)
+  * [Devenv](guides/installation/devenv.md)
```
Comment (fschmtt) [inline SUMMARY.md:118]: The devenv guide should also be included on the overview page

**Review by Christian-Rades — APPROVED** [review]

File: guides/installation/devenv.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+```
+
+Before booting up your development environment, configure Cachix to use Shopware's cache:
+
+```shell
+cachix use shopware
+```
+
+By default, `shopware/platform` uses unfree software like Blackfire.
+To be able to use unfree software, you have to allow that:
```
Comment (sushmangupta) [inline guides/installation/devenv.md:99]:
```suggestion
By default, `shopware/platform` uses unfree software like Blackfire. To be able to use unfree software, you have to *allow* it:
```

File: guides/installation/devenv.md [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+<!-- markdown-link-check-disable-next-line -->
+{% embed url="https://devenv.sh/getting-started/" caption="Getting started - devenv.sh" %}
+
+### Shopware
+
+Depending on whether you want to set up a fresh Shopware project or contribute to the Shopware core, choose between:
+
+{% tabs %}
+{% tab title="Symfony Flex" %}
+If you are already using Symfony Flex, you can require a Composer package to get a basic devenv configuration:
```
Comment (sushmangupta) [inline guides/installation/devenv.md:117]:
```suggestion
If you are already using Symfony Flex, you require a composer package to get a basic devenv configuration:
```

File: guides/installation/devenv.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+Since the environment is described via a `devenv.nix` file committed to version control, you can now boot up the environment:
+
+```shell
+devenv up
+```
+
+{% hint style="info" %}
+Make sure that the ports for the services are not already in use or the command will fail.
+{% endhint %}
```
Comment (sushmangupta) [inline guides/installation/devenv.md:144]:
```suggestion
{% hint style="warning" %}
Make sure that the ports for the services are not already in use, or else the command will fail.
{% endhint %}
```

File: guides/installation/devenv.md [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+{% code title="<PROJECT_ROOT>/.env" %}
+
+```dotenv
+DATABASE_URL="mysql://shopware:shopware@127.0.0.1:3306/shopware?sslmode=disable&charset=utf8mb4"
+```
+
+{% endcode %}
+
+Still inside the project directory, run the following command to initialize Shopware:
```
Comment (sushmangupta) [inline guides/installation/devenv.md:156]:
```suggestion
Inside the project directory, run the following command to initialize Shopware:
```

File: guides/installation/devenv.md [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+
+```shell
+composer setup
+```
+
+### Direnv
+
+If you wish to seamlessly switch between multiple development environments which use devenv we recommend to install [direnv](https://direnv.net/).
```
Comment (sushmangupta) [inline guides/installation/devenv.md:164]:
```suggestion
If you wish to switch between multiple development environments which use devenv seamlessly, we recommend installing [direnv](https://direnv.net/).
```

File: guides/installation/devenv.md [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+```bash
+direnv allow
+```
+
+<!-- markdown-link-check-disable-next-line -->
+{% embed url="https://devenv.sh/automatic-shell-activation/" caption="Automatic Shell Activation - devenv.sh" %}
+
+## Default services
+
+Here's an overview of services Shopware provides by default and how you can access them:
```
Comment (sushmangupta) [inline guides/installation/devenv.md:245]:
```suggestion
Here is an overview of services Shopware provides by default and how you can access them:
```

File: guides/installation/devenv.md [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+|------------------|------------------------------------------------|
+| MySQL            | `mysql://shopware:shopware@127.0.0.1:3306`     |
+| Caddy            | [http://localhost:8000](http://localhost:8000) |
+| Adminer          | [http://localhost:9080](http://localhost:9080) |
+| Mailhog (SMTP)   | `smtp://127.0.0.1:1025`                        |
+| Mailhog (Web UI) | [http://localhost:8025](http://localhost:8025) |
+
+## Customize your setup
+
+To customize the predefined services to match your needs, e.g. changing the virtual host, database name or environment variables, you can create `devenv.local.nix` to override the service definitions.
```
Comment (sushmangupta) [inline guides/installation/devenv.md:257]:
```suggestion
To customize the predefined services to match your needs, e.g., changing the virtual host, database name, or environment variables, you can create `devenv.local.nix` to override the service definitions.
```

File: guides/installation/devenv.md [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+
+### How do I clean up devenv?
+
+Periodically run `devenv gc` to remove orphaned services, packages and processes and free up disk space.
+
+### How do I access the database?
+
+The MySQL service is exposed under its default port `3306`, see [Default services](#default-services).
+
+Be aware that you cannot connect using the `localhost` socket, you must use `127.0.0.1` instead.
```
Comment (sushmangupta) [inline guides/installation/devenv.md:359]:
```suggestion
Be aware that you cannot connect using the `localhost` socket. Instead, you must use `127.0.0.1`.
```

**Review by sushmangupta — CHANGES_REQUESTED** [review]

**Review by Christian-Rades — APPROVED** [review]

**Review by Christian-Rades — APPROVED** [review]

**Pull request merged**